packages/flame/lib/src/effects2/rotate_effect.dart

```diff
@@ -0,0 +1,50 @@
+import 'effect_controller.dart';
+import 'transform2d_effect.dart';
+
+/// Rotate a component around its anchor.
+///
+/// Two constructors are provided:
+///   - [RotateEffect.by] will rotate the target by the specified `angle`
+///     relative to its orientation at the onset of the effect. For example,
+///     rotating by `angle = tau/4` will turn the component 90° clockwise
+///     relative to its initial direction;
+///   - [RotateEffect.to] will rotate the target to the fixed orientation
+///     specified by the `angle`. For example, rotating to `angle = tau/4` will
+///     turn the component to look East regardless of its initial bearing.
+///
+/// This effect applies incremental changes to the component's angle, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class RotateEffect extends Transform2DEffect {
+  RotateEffect.by(double angle, EffectController controller)
+      : _angle = angle,
+        super(controller);
+
+  factory RotateEffect.to(double angle, EffectController controller) {
+    return _RotateToEffect(angle, controller);
+  }
+
+  /// The magnitude of the effect: how much the target should turn as the
+  /// progress goes from 0 to 1.
+  double _angle;
+
+  @override
+  void apply(double progress) {
+    final dProgress = progress - previousProgress;
+    target.angle += _angle * dProgress;
+    super.apply(progress);
+  }
+}
+
+class _RotateToEffect extends RotateEffect {
```

Comment:
Does this really need to be done as a separate class, can't this be done in the factory?

Reply:
This is a private class, invoked from the factory constructor `RotateEffect.to()`.
Can it be merged with the "main" RotateEffect class? It can, but it'll bring an extra variable `_destinationAngle`, a boolean parameter `_isRelative`, and an override for `onStart()` which is not needed by the main class. I don't think this would improve code clarity...

Also, I've been thinking about eventually adding another flavor of the rotate effect: `RotateEffect.towards()`, which would be similar to `RotateEffect.to()`, except that it would perform the rotation along the shortest path (for example, rotating from 359º to 0º should not make a full circle, but instead choose a 1º arc). It would be easier to implement such class as another override, instead of converting `_isRelative` into an enum.